crates/sui-types/src/transaction.rs

```diff
@@ -150,6 +152,14 @@ impl WithdrawalTypeArg {
             }
         }
     }
+
+    /// Get the type parameter of the withdrawal type argument,
+    /// e.g. `Balance<T>` -> `T`
+    pub fn get_type_param(&self) -> anyhow::Result<TypeTag> {
+        match self {
+            WithdrawalTypeArg::Balance(type_param) => type_param.to_type_tag(),
+        }
+    }
```

Comment:
This might be a bit more future proof, just in that we don't want to check the deny list for non-balance types, and as such, we can be a bit more intentional here

```
pub fn balance_type_tag() -> anyhow::Result<Option<TypeTag>> {
    match self {
        WithdrawalTypeArg::Balance(type_param) => Some(type_param.to_type_tag()?),

    }
}
```